### PR TITLE
Use jar-dependencies jruby library

### PR DESCRIPTION
**Pull request by ebuildy**

### CODE REVIEW

**Comment by garyelephant** [comment]
@ebuildy What does `lib/logstash-filter-geoip_jars.rb` do and why you ignore it in git commit ?

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ Gemfile.lock
 .bundle
 vendor
 .idea
+lib/com
+lib/logstash-filter-geoip_jars.rb
```
Comment (suyograo) [inline .gitignore]: @ebuildy I dunno if we should be excluding this. This file does all the magic of loading jars

**Comment by suyograo** [comment]
@ebuildy did you make sure all tests pass with this change? 

`bundle install`
`bundle exec rspec`

**Comment by suyograo** [comment]
@ebuildy also we may want to do this: https://github.com/mkristian/jar-dependencies#vendoring-your-jars-before-packing-the-jar

and add a `rake install_jars` task

File: .gitignore (hunk 1 above)
Comment (ebuildy) [inline .gitignore]: I don't know, this is a auto-generated file that will be overriden by jar-dependencies rake task after downloaded jars from Maven.

**Comment by ebuildy** [comment]
@suyograo I did it, but I found that running `plugin install --no-verify` do the trick. 

I searched for some examples in logstash-plugins repositories, found this : https://github.com/logstash-plugins/logstash-input-kafka/commit/861e834fba5d6874e821b38cd1c39d91edc5c6bd and this https://github.com/logstash-plugins/logstash-output-neo4j.

But I didn't find where logstash do this ;-( Do you want me to still add this?

Thanks you,

**Comment by suyograo** [comment]
@ebuildy yes we need those files. `lib/logstash-filter-geoip_jars.rb` loads the jars from classpath.

Can you also add the rake task?

**Comment by ebuildy** [comment]
Ha , jars must be  in  git  ?

Thanks,

**Comment by ebuildy** [comment]
I have added jars and rake file (but I tested w/o and logstash seems to install everything needed).

**Comment by suyograo** [comment]
@ebuildy Sorry, just the file `lib/logstash-filter-geoip_jars.rb`. You are correct that `lib/*` needs to be excluded.

So in `.gitignore`: `.idea` `lib/com` can be excluded

**Comment by ebuildy** [comment]
No problem, this should do the trick!

Tested with logstash 2.3, works fine, library fasterxml is safely ignored and the one provided by logstash is used.

**Comment by garyelephant** [comment]
Thanks @ebuildy , I will merge this pull request.

**Comment by garyelephant** [comment]
@suyograo @ebuildy Should I delete `lib/com` dir ? Looks like @ebuildy forgot it.

**Comment by suyograo** [comment]
@garyelephant yes please delete `lib/com` dir, we don't want all those jars checked in :). Thanks @ebuildy

**Pull request merged**